### DISCUSSION TITLE
How MST can cancelling previously async calls?

### DISCUSSION THREAD

**Discussion by tonyfung99** [mobxjs/mobx-state-tree#1715] [Q&A] [answered]
Hello guys, 

I would like to know is there any way like `switchMap` in RxJS & `takeLatest` in redux-saga? I want to drop all the expired async actions and only keep the latest triggered async actions. 

I searched there's some way to handle in axios cancel token, but I would like to see is there a more generic way to handle async behaviour in MST, Thanks. 

https://stackoverflow.com/questions/67467875/is-there-any-way-to-cancel-previous-async-action-in-mobx-state-tree-similar-to

**Comment by EmilTholin** [comment, marked as answer]
Hi @tonyfung99!

There is [a WIP PR](https://github.com/mobxjs/mobx-state-tree/pull/691) discussing this topic.
